plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -1094,14 +1096,18 @@ def disableTimer(self) -> None:
             self._change_timer.timeout.disconnect(self.slice)
 
     def _onPreferencesChanged(self, preference: str) -> None:
-        if preference != "general/auto_slice" and preference != "info/send_engine_crash":
+        if preference != "general/auto_slice" and preference != "info/send_engine_crash" and preference != "info/anonymous_engine_crash_report":
             return
         if preference == "general/auto_slice":
             auto_slice = self.determineAutoSlicing()
             if auto_slice:
                 self._change_timer.start()
         elif preference == "info/send_engine_crash":
-            os.environ["use_sentry"] = "1" if CuraApplication.getInstance().getPreferences().getValue("info/send_engine_crash") else "0"
+            os.environ["USE_SENTRY"] = "1" if CuraApplication.getInstance().getPreferences().getValue("info/send_engine_crash") else "0"
+        elif preference == "info/anonymous_engine_crash_report":
+            account = CuraApplication.getInstance().getCuraAPI().account
+            if account and account.isLoggedIn and not CuraApplication.getInstance().getPreferences().getValue("info/anonymous_engine_crash_report"):
```

Comment:
We could maybe handle this the same as the optional `project_name` and send it over with the slice message and check when actually pressing slice. But that will remove the environmental option which I kinda like for future cli work.

Reply:
It can also be done in the StartSliceJob, but by setting the env variable

Reply:
So this is apparently not possible, because the process is started way before the Slice button is pressed, so settting the environment variable after the process has been started will only work for the next start. That gives us a few options:
1. Leave it as is, will not always work
2. Update the env variable on user login/logout so that it will at least be ok for the next slice
3. Add it to the slice message (which will also work when using --external_backend)

I would vote for 3. but you have a more experienced look

Reply:
3 is fine by me